### ISSUE TITLE
Bump versions of google-cloud-{core,trace} from May 2017?

### ISSUE THREAD

**Issue by xlfe** [GoogleCloudPlatform/opentelemetry-operations-java#101]
I'm attempting to migrate to otel on GKE and I am trying to use this library for trace exports. I'm running into problems though because my project also uses a jaeger exporter which is using grpc-protobuf 1.34.2

It seems that you have pinned googleCloudVersion at 1.0.2 from May 2017 which is referencing grpc-protobuf 1.25.

I have tried including the latest versions, and can get around the grpc-incompatibility but I'm now running into incompatibilities in the new versions of google-cloud-* because (at least one) API has changed.
 
```
java.lang.NoClassDefFoundError: com/google/cloud/trace/v2/TraceServiceSettings 
at com.google.cloud.opentelemetry.trace.TraceExporter.createWithCredentials(TraceExporter.java:85) 
at com.google.cloud.opentelemetry.trace.TraceExporter.createWithConfiguration(TraceExporter.java:62) 
at com.google.cloud.opentelemetry.trace.TraceExporter.createWithDefaultConfiguration(TraceExporter.java:48)
...
```

I imagine there are other breaking changes. Any chance you can update this library to use the latest release of google-cloud-core, google-cloud-trace?

Also, I note that your 0.14.0 release in maven still has the otel dependencies versions set at 1.0.0 even though in this repo it looks like it should be set to 1.0.1 ?

**Comment by jsuereth** [comment]
Haven't released against 1.0.1 yet.  There's a few features we'd like to add to the trace-exporter prior to that, but I'll likely cut one earlier with this bump.


QQ:   Would you rather have us "shade" our GRPC/Cloud API jars?  This would make the exporter itself larger, but could help alleviate this kind of problem in the future (and help with users in the opposite scenario as you, i.e. a stale version of gRPC from some other component).

**Comment by xlfe** [comment]
Thanks @jsuereth 

Re Shading - that's an interesting one. It seems like a trade-off?

On one hand, you could add shading to your releases and they would be larger. 

You're not doing that yet, though, I think? When I unzip the 0.14 jar from maven it just has the Class files

```
Archive:  ../exporter-trace-0.14.0.jar
   creating: META-INF/
  inflating: META-INF/MANIFEST.MF    
   creating: com/
   creating: com/google/
   creating: com/google/cloud/
   creating: com/google/cloud/opentelemetry/
   creating: com/google/cloud/opentelemetry/trace/
  inflating: com/google/cloud/opentelemetry/trace/TraceTranslator$1.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceConfiguration.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceTranslator$2.class  
  inflating: com/google/cloud/opentelemetry/trace/CloudTraceClientImpl.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceExporter.class  
  inflating: com/google/cloud/opentelemetry/trace/AutoValue_TraceConfiguration$Builder.class  
  inflating: com/google/cloud/opentelemetry/trace/AutoValue_TraceConfiguration$1.class  
  inflating: com/google/cloud/opentelemetry/trace/CloudTraceClient.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceConfiguration$Builder.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceVersions.class  
  inflating: com/google/cloud/opentelemetry/trace/AutoValue_TraceConfiguration.class  
  inflating: com/google/cloud/opentelemetry/trace/TraceTranslator.class  
  inflating: com/google/cloud/opentelemetry/trace/version.properties  
```

On the other hand, without shading - each time otel has a new release, you could also have a release and tie it to the latest google-cloud-* as well. That way, anyone using your library can choose the release they need based on which otel version they want and the other dependencies. As long as they're not in the opposite situation to me (where they have older versions of gRPC for example).

So on that basis, it feels like the only reason you'd need shading is if (as it is currently) some of your dependencies are so old as to be incompatible for anything with recent versions?

My project is greenfields (ie we don't have lots of legacy code) so we can stay up to date with new releases. I'd prefer the non-shaded option with up-to-date dependences, but for projects that have old dependencies, I can see why shading makes sense too. 

Although, you could make the argument older projects won't be moving to otel immediately (eg more likely to be staying with open-census).
